#### 🚀🪐 Список будет постепенно обновляться новыми метриками 🌌🛸

<font color='red' size=5>**Представленные метрики перечислены по принципу:**</font>
--
 1. Название (аббревиатура, варианты на английском и на русском)
 2. Описание (короткое описание)
 3. Формула (будут создаваться с помощью LaTex)
 4. Пример (для расчетов будут создаваться датафреймы с рандомными значениями с помощью Pandas/Numpy)
 5. Интерпретация результата (что даёт результат?)


-----

<font color='fuchsia' size=5>ROI (Return on Investment — возврат инвестиций)</font>
===
метрика показывает возврат инвестиций от вложенных денег в маркетинг
--
<font color='green' size=5>формула:</font>
    $\LARGE ROI=\frac{(\text{Доход - Расходы})}{\text{Расходы на маркетинг}}*100 \%$

<font color='brown' size=5>пример:</font>

In [1]:
import pandas as pd
import numpy as np

np.random.seed(0)
data = {'Месяц':['Январь','Февраль','Март','Апрель','Май','Июнь','Июль','Август','Сентябрь','Октябрь','Ноябрь','Декабрь'], 
        'Доходы': np.random.randint(100_000, 150_000, size=12),
        'Расходы': np.array([100_000]*12),
        'Расходы на маркетинг': np.random.choice([5000,10000], size=12),
        'ROI (в %)': np.arange(12)} 
df = pd.DataFrame(data)
df["Расходы"] = np.where(df["Расходы на маркетинг"] == 5000, df["Расходы"]+5000, df["Расходы"]+10000)
df['ROI (в %)'] = ((df['Доходы']-df['Расходы'])/df['Расходы на маркетинг']*100).round()
df

Месяц  Доходы  Расходы  Расходы на маркетинг  ROI (в %)
0     Январь  102732   105000                  5000      -45.0
1    Февраль  143567   105000                  5000      771.0
2       Март  142613   105000                  5000      752.0
3     Апрель  145891   105000                  5000      818.0
4        Май  121243   110000                 10000      112.0
5       Июнь  130403   105000                  5000      508.0
6       Июль  132103   110000                 10000      221.0
7     Август  141993   110000                 10000      320.0
8   Сентябрь  120757   105000                  5000      315.0
9    Октябрь  146884   105000                  5000      838.0
10    Ноябрь  114935   110000                 10000       49.0
11   Декабрь  115430   110000                 10000       54.0

Если метрика ROI меньше 0% — вы понесли убытки. Если равен 0% — вложения окупились, но прибыли нет. Если больше 0% — вы получили прибыль
--

---

<font color='fuchsia' size=5>CRR (Customer Retention Rate — коэффициент удержания клиентов)</font>
===
метрика показывает процент клиентов, которые возвращаются для совершения повторных покупок
--
<font color='green' size=5>формула:</font>
    $\large CRR=\frac{(\text{Количество клиентов в конце периода - Количество новых клиентов за период})}{\text{Количество клиентов в начале расчетного периода}}*100 \%$

**формула расчета "Количество клиентов в конце периода":**
$\scriptsize \text{Количество клиентов в конце периода} =\text{Количество клиентов в начале периода + Количество новых клиентов за период - Количество неактивных клиентов за период}$

<font color='brown' size=5>пример:</font>

In [2]:
np.random.seed(0)
data_crr = {'CustomerID': np.random.randint(1, 20, size=20), 
        'CustomerPurchases': np.random.randint(0, 20, size=20),
        'NewCustomerIndicator': np.array([0,1,0,0,0,0,0,1,1,0,0,0,0,0,1,0,0,0,0,0])} 

df_crr = pd.DataFrame(data_crr)
df_crr.index = np.arange(1, len(df_crr)+1)
df_crr["CustomerPurchases"] = np.where(df_crr["CustomerID"].isin([10,18,9]), 
                                        ~df_crr["CustomerPurchases"] == 0, df_crr["CustomerPurchases"])
# 18,10,9 - неактивные клиенты
# 5,19,16,15 - новые клиенты

df_crr

CustomerID  CustomerPurchases  NewCustomerIndicator
1           13                 19                     0
2           16                 16                     1
3            1                 19                     0
4            4                  5                     0
5            4                 15                     0
6            8                 15                     0
7           10                  0                     0
8           19                 18                     1
9            5                  3                     1
10           7                 17                     0
11          13                 19                     0
12           2                 19                     0
13           7                 19                     0
14           8                 14                     0
15          15                  7                     1
16          18                  0                     0
17           6                  1                     0
18          14                  9                     0
19           9                  0                     0
20          10                  0                     0

$\large 11_\text{(Количество клиентов в начале периода)} - 3_\text{(Количество неактивных клиентов)} + 4_\text{(Количество новых клиентов)} = 12_\text{(Количество клиентов в конце периода)}$

$\large CRR=\frac{(\text{12 - 4})}{\text{11}}*100 \% \thickapprox 72\%$

Метрика CRR, демонстрирует эффективность удержания клиентов. </br>Чем выше показатель, тем лучше работает стратегия удержания 
-- 

---

<font color='fuchsia' size=5>CR (Churn rate — коэффициент оттока клиентов) </font>
===
метрика показывает процент оттока клиентов и вместе с тем - выручки
--
<font color='green' size=5>формула:</font>
    $\large CR=\frac{(\text{Количество клиентов в начале периода - Количество клиентов в конце периода})}{\text{Количество клиентов в начале периода}}*100 \%$

<font color='brown' size=5>пример:</font>

Количество клиентов в начале периода (например за месяц) - **1500 клиентов** </br>
Количество клиентов в конце периода (например за месяц) - **1600 клиентов** </br>

$\large CR = \frac{(1500-1600)}{1500}*100 \% \thickapprox -7 \%$

Допустимые значения Churn rate отличаются в зависимости от сферы деятельности предприятия. Норма для SaaS сервисов от 4 до 6%, для продуктов ежедневного спроса от 10 до 11%, для онлайн и цифрового телевидения от 11 до 13%. Отдельные нормы churn rate для новичков рынка и стартапов. Там показатель оттока может достигать 20%. </br>Отрицательное значение свидетельствует об отсутствии оттока клиентов
--